src/test/java/seedu/address/logic/commands/EditCommandTest.java

```diff
@@ -0,0 +1,192 @@
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_RECIPE;
+import static seedu.address.testutil.TypicalRecipes.getTypicalRecipeBook;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.EditCommand.EditRecipeDescriptor;
+import seedu.address.model.Model;
+import seedu.address.model.RecipeBook;
+import seedu.address.model.RecipeBookModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.recipe.Recipe;
+import seedu.address.testutil.EditRecipeDescriptorBuilder;
+import seedu.address.testutil.RecipeBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for EditCommand.
+ */
+public class EditCommandTest {
+
+    private Model model = new RecipeBookModelManager(getTypicalRecipeBook(), new UserPrefs());
+
+    @Test
+    public void execute_allFieldsSpecifiedUnfilteredList_success() {
+        Recipe editedRecipe = new RecipeBuilder().build();
+        EditRecipeDescriptor descriptor = new EditRecipeDescriptorBuilder(editedRecipe).build();
+        EditCommand editCommand = new EditCommand(INDEX_FIRST_RECIPE, descriptor);
+
+        String expectedMessage = String.format(EditCommand.MESSAGE_EDIT_RECIPE_SUCCESS, editedRecipe);
+
+        Model expectedModel = new RecipeBookModelManager(new RecipeBook(model.getRecipeBook()), new UserPrefs());
+        expectedModel.setRecipe(model.getFilteredRecipeList().get(0), editedRecipe);
+
+        assertCommandSuccess(editCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_someFieldsSpecifiedUnfilteredList_success() {
+        Index indexLastRecipe = Index.fromOneBased(model.getFilteredRecipeList().size());
+        Recipe lastRecipe = model.getFilteredRecipeList().get(indexLastRecipe.getZeroBased());
+
+        RecipeBuilder recipeInList = new RecipeBuilder(lastRecipe);
+        Recipe editedRecipe = recipeInList
+                .withName(VALID_NAME_CHICKEN_CHOP)
+                .withCompletionTime(VALID_COMPLETION_TIME_CHICKEN_CHOP)
+                .withServingSize(VALID_SERVING_SIZE_CHICKEN_CHOP)
+                .withIngredients(VALID_INGREDIENT_CHICKEN_CHICKEN_CHOP.toString(),
+                        VALID_INGREDIENT_POTATO_CHICKEN_CHOP.toString())
+                .withSteps(VALID_STEP_1_CHICKEN_CHOP.toString())
+                .withTags(VALID_TAG_CHICKEN_CHOP).build();
+
+        EditRecipeDescriptor descriptor = new EditRecipeDescriptorBuilder()
+                .withName(VALID_NAME_CHICKEN_CHOP)
+                .withCompletionTime(VALID_COMPLETION_TIME_CHICKEN_CHOP)
+                .withServingSize(VALID_SERVING_SIZE_CHICKEN_CHOP)
+                .withIngredients(VALID_INGREDIENT_CHICKEN_CHICKEN_CHOP,
+                        VALID_INGREDIENT_POTATO_CHICKEN_CHOP)
+                .withSteps(VALID_STEP_1_CHICKEN_CHOP)
```

Comment:
Refer to the previous comment/suggestion, only implement either one and stick with it

```suggestion
                .withIngredients(VALID_INGREDIENT_CHICKEN_CHICKEN_CHOP.toString(),
                        VALID_INGREDIENT_POTATO_CHICKEN_CHOP.toString())
                .withSteps(VALID_STEP_1_CHICKEN_CHOP.toString())
```